### PR TITLE
Split GitHub Actions workflows

### PR DESCRIPTION
**Pull request by toshihikoyanase**
This PR is related to https://github.com/optuna/optuna/issues/2654.

This PR splits the CI job of Optuna examples.
The original workflow file is [optuna/.github/workflows/examples.yaml](https://github.com/optuna/optuna/blob/ed5086f8d36655d706c26e603154ebda9193b3df/.github/workflows/examples.yml)

The changes are as follows:
- Create subdirectories for catboost, dask_ml, haiku, ray, skimage and tensorboard
- Add `requirements.txt` to mange dependencies
- Update links in README.md
- Add GitHub Actions workflow files for CI

### CODE REVIEW

File: .github/workflows/allennlp.yml
Comment (hvy) [inline .github/workflows/allennlp.yml:17]: Seems like there's a way to create a matrix over `runs-on` in which case we could iterate over prebuilt optuna official docker images https://github.community/t/create-matrix-with-multiple-os-and-env-for-each-one/16895/2. But this also adds another level of complexity so it's just a comment for the record.

File: .github/workflows/pytorch.yml
Comment (toshihikoyanase) [reply .github/workflows/pytorch.yml:66]: Thank you for pointing it out. I'll fix it!

Comment (toshihikoyanase) [reply .github/workflows/pytorch.yml]: Fixed in https://github.com/optuna/optuna-examples/pull/1/commits/ed73921a5c4527827088dccb728e90486a2c0890.

**Comment by toshihikoyanase** [comment]
The CI jobs passed. PTAL.

**Comment by hvy** [comment]
Quickly checked the examples not covered by CI after this PR.

```
kubernetes/
multi_objective/
samplers/
visualization/
enqueue_trial.py
max_trials_callback.py
quadratic_simple.py
quickstart.ipynb
rapids_simple.py
ray_joblib.py
simple_pruning.py
skimage_lbp_simple.py
```

File: .github/workflows/allennlp.yml
Comment (hvy) [inline .github/workflows/allennlp.yml:19]: In the future, it might make sense to remove this check given https://github.com/optuna/optuna/pull/2656. I also noticed that PyTorch has a disclaimer at the top of their README to disable daily when forking at the user side.

```
WARNING: if you fork this repo, github actions will run daily on it. To disable this, go to /examples/settings/actions and Disable Actions for this repository
```

https://github.com/pytorch/examples

**Review by hvy — APPROVED** [review]
LGTM!

**Comment by toshihikoyanase** [comment]
We'll relax the review policy. The [optuna/optuna](https://github.com/optuna/oputna) requires two approval reviews, but this repository requires a single approval. If we find issues after merging PRs, we'll work on them in follow-up PRs.
So, let me merge this PR.

**Pull request merged**
